### DISCUSSION TITLE
debug with tmux

### DISCUSSION THREAD

**Discussion by bl4ckhyun** [Gallopsled/pwntools#2292] [Q&A]
```
from pwn import *
...
p = gdb.debug('./left_right_v2')
...
context.terminal = ["tmux", "splitw", "-v"]
```

I used that command in my Python script, but tmux splits horizontally instead of vertically. Is there a reason why?

**Comment by peace-maker** [comment]
The `context.terminal` option as to be specified before you launch a process in a new terminal. So try moving that line up before the `gdb.debug` call.

**Reply by bl4ckhyun** [reply]
```
//ex.py
 from pwn import *
 
 context.terminal = ["tmux", "splitw", "-v"]
 p = gdb.debug('./left_right_v2')
```
i changed my python code like your comment

```
root$ tmux  //run tmux
root$python3 ex.py //run the python code after excute tmux  
```

i follow your comment but nothing changed : (

**Comment by RocketMaDev** [comment]
tmux's behavior is kinda weird. Use `-t` to vertically split the terminal.
